sdk/samples/core/mosquitto_multiple_rpc_sample.c

```diff
@@ -0,0 +1,629 @@
+      = { .correlation_id = pending_server_command.correlation_id,
+          .error_message = AZ_SPAN_FROM_STR("Command Server timeout"),
+          .response_topic = pending_server_command.response_topic,
+          .request_topic = pending_server_command.request_topic,
+          .status = AZ_MQTT5_RPC_STATUS_TIMEOUT,
+          .response = AZ_SPAN_EMPTY,
+          .content_type = AZ_SPAN_EMPTY };
+  if (az_result_failed(az_mqtt5_rpc_server_execution_finish(&rpc_server_policy, &return_data)))
+  {
+    printf(LOG_APP_ERROR "Failed sending execution response to HFSM\n");
+    return;
+  }
+
+  pending_server_command.content_type = AZ_SPAN_FROM_BUFFER(server_content_type_buffer);
+  pending_server_command.request_topic = AZ_SPAN_FROM_BUFFER(server_request_topic_buffer);
+  pending_server_command.correlation_id = AZ_SPAN_EMPTY;
+
+#ifdef _WIN32
+  return AZ_ERROR_DEPENDENCY_NOT_PROVIDED;
+#else
+  if (0 != timer_delete(timer))
+  {
+    printf(LOG_APP_ERROR "Failed destroying timer\n");
+    return;
+  }
+#endif
+}
+
+/**
+ * @brief Start a timer
+ */
+AZ_INLINE az_result start_timer(void* callback_context, int32_t delay_milliseconds)
+{
+#ifdef _WIN32
+  return AZ_ERROR_DEPENDENCY_NOT_PROVIDED;
+#else
+  sev.sigev_notify = SIGEV_THREAD;
+  sev.sigev_notify_function = &timer_callback;
+  sev.sigev_value.sival_ptr = &callback_context;
+  if (0 != timer_create(CLOCK_REALTIME, &sev, &timer))
+  {
+    return AZ_ERROR_ARG;
+  }
+
+  // start timer
+  trigger.it_value.tv_sec = delay_milliseconds / 1000;
+  trigger.it_value.tv_nsec = (delay_milliseconds % 1000) * 1000000;
+
+  if (0 != timer_settime(timer, 0, &trigger, NULL))
+  {
+    return AZ_ERROR_ARG;
+  }
+#endif
+
+  return AZ_OK;
+}
+
+/**
+ * @brief Stop the timer
+ */
+AZ_INLINE az_result stop_timer()
+{
+#ifdef _WIN32
+  return AZ_ERROR_DEPENDENCY_NOT_PROVIDED;
+#else
+  if (0 != timer_delete(timer))
+  {
+    return AZ_ERROR_ARG;
+  }
+#endif
+
+  return AZ_OK;
+}
+
+/**
+ * @brief Function that does the actual command execution
+ * @note Needs to be modified for your solution
+ */
+az_mqtt5_rpc_status execute_command()
+{
+  // for now, just print details from the command
+  printf(
+      LOG_APP "Executing command from topic: %s\n",
+      az_span_ptr(pending_server_command.request_topic));
+  return AZ_MQTT5_RPC_STATUS_OK;
+}
+
+/**
+ * @brief Check if there is a pending command and execute it. On completion, if the command hasn't
+ * timed out, send the result back to the hfsm
+ * @note Result to be sent back to the hfsm needs to be modified for your solution
+ */
+az_result check_for_commands_to_execute()
+{
+  if (az_span_ptr(pending_server_command.correlation_id) != NULL)
+  {
+    // copy correlation id to a new span so we can compare it later
+    uint8_t copy_buffer[az_span_size(pending_server_command.correlation_id)];
+    az_span correlation_id_copy
+        = az_span_create(copy_buffer, az_span_size(pending_server_command.correlation_id));
```

Comment:
```suggestion
        = az_span_create(copy_buffer, correlation_id_size(pending_server_command.correlation_id));
```

Reply:
Let me know if the change I pushed for this was what you had in mind